plugin/trino-exchange/src/main/java/io/trino/plugin/exchange/s3/ExchangeS3Config.java

```diff
@@ -119,4 +122,17 @@ public ExchangeS3Config setS3UploadPartSize(DataSize s3UploadPartSize)
         this.s3UploadPartSize = s3UploadPartSize;
         return this;
     }
+
+    @NotNull
+    public StorageClass getStorageClass()
+    {
+        return storageClass;
+    }
+
+    @Config("exchange.s3.storage-class")
+    public ExchangeS3Config setStorageClass(StorageClass storageClass)
```

Comment:
please do not use enum from external library directly as config parameter setter, to have full control over what we can set in configuration file.

Reply:
I was also thinking about that. Here's my thoughts. I don't think they are going to remove an existing option, as it may break the compatibility. And when they add a new option we probably want it to also be available.

Reply:
Yeah. But still (I think) the common case is to either have our own enum which maps to enum from an external library. Or make `setSomething` consume a `String` and do `switch...case` in the setter body to map to enum fields.
Adding a new option to our mapping if AWS adds a new enum value is a minimalistic effort. And we have better control over product surface, and also config values naming.

Reply:
As a reference trino-hive uses our own enum https://github.com/trinodb/trino/blob/4ae3121be176a3c70a6917414abbcaa7a155b78c/plugin/trino-hive/src/main/java/io/trino/plugin/hive/s3/TrinoS3StorageClass.java#L20

Reply:
There are other places in the codebase when the library enum is used directly:

https://github.com/trinodb/trino/blob/193db65de2c24fb8216607d01502226e74a65bef/plugin/trino-cassandra/src/main/java/io/trino/plugin/cassandra/CassandraClientConfig.java#L50
https://github.com/trinodb/trino/blob/c647c317405565ebd84599953dbf3cfca42affec/plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaSecurityConfig.java#L30

The `TrinoS3StorageClass` might not be the best example as it does two things:

1. Translates available values from lower case to upper case
2. Limits the available options

The new version of the AWS libraries already uses values in upper case. I'm not sure if there's a value of limiting the available options. While it is rather unlikely that we will ever try to use `GLACIER`, we may still want to play around with reduced concurrency and reduced availability options. So I don't want to limit the available options just yet. Also using the library enum directly will make any new options available without any modifications and it's rather unlikely for the existing options to be removed as I'm pretty sure AWS developers won't be willing to break compatibility.

Thus I'm not sure if there's a significant practical reason of adding a proxy.